src/gfx_layout.cpp

```diff
@@ -37,6 +42,54 @@
 /** Cache of ParagraphLayout lines. */
 std::unique_ptr<Layouter::LineCache> Layouter::linecache;
 
+class RuntimeMissingGlyphSearcher : public MissingGlyphSearcher {
+	std::array<std::set<char32_t>, FS_END> glyphs{};
+	bool any_missing = false; ///< Set if any glyph
+public:
+	RuntimeMissingGlyphSearcher() : MissingGlyphSearcher({FS_NORMAL, FS_SMALL, FS_LARGE}) {}
+
+	FontLoadReason GetLoadReason() override { return FontLoadReason::MissingFallback; }
+
+	inline void Insert(FontSize fs, char32_t c)
+	{
+		this->glyphs[fs].insert(c);
+		this->any_missing = true;
+	}
+
+	std::set<char32_t> GetRequiredGlyphs(FontSizes fontsizes) override
+	{
+		std::set<char32_t> r;
+		for (FontSize fs : fontsizes) {
+			r.merge(this->glyphs[fs]);
+		}
+		return r;
+	}
+
+	const IntervalTimer<TimerWindow> check_missing_glyphs_interval{std::chrono::milliseconds(250), [this](auto)
```

Comment:
Isn't a `TimeoutTimer<TimerWindow>` a better fit conceptually? After all, with `any_missing` you implement something similar.

You can remove `any_missing` and instead of `this->any_missing = true` you do `this->check_missing_glyphs_interval.Reset();`. You might want to rename the timer as well in that case.